cf-testd/cf-testd.c

```diff
@@ -0,0 +1,492 @@
+    if (IsPendingTermination())
+    {
+        Log(LOG_LEVEL_VERBOSE, "Server must exit, closing connection");
+        return false;
+    }
+
+    // See BusyWithNewProtocol() in server_tls.c for how to add other commands
+    switch (GetCommandNew(recvbuffer))
+    {
+    case PROTOCOL_COMMAND_QUERY:
+    {
+        char query[256], name[128];
+        int ret1 = sscanf(recvbuffer, "QUERY %255[^\n]", query);
+        int ret2 = sscanf(recvbuffer, "QUERY %127s", name);
+        if (ret1 != 1 || ret2 != 1)
+        {
+            return CFTestD_ProtocolError(conn, recvbuffer, sendbuffer);
+        }
+
+        if (CFTestD_GetServerQuery(conn, recvbuffer, encrypted))
+        {
+            return true;
+        }
+
+        break;
+    }
+    case PROTOCOL_COMMAND_BAD:
+    default:
+        Log(LOG_LEVEL_WARNING, "Unexpected protocol command: %s", recvbuffer);
+    }
+
+    return CFTestD_ProtocolError(conn, recvbuffer, sendbuffer);
+}
+
+static ServerConnectionState *CFTestD_NewConn(ConnectionInfo *info)
+{
+#if 1
+    /* TODO: why do we do this ?  We fail if getsockname() fails, but
+     * don't use the information it returned.  Was the intent to use
+     * it to fill in conn->ipaddr ? */
+    struct sockaddr_storage addr;
+    socklen_t size = sizeof(addr);
+
+    if (getsockname(ConnectionInfoSocket(info), (struct sockaddr *)&addr, &size) == -1)
+    {
+        Log(LOG_LEVEL_ERR,
+            "Could not obtain socket address. (getsockname: '%s')",
+            GetErrorStr());
+        return NULL;
+    }
+#endif
+
+    ServerConnectionState *conn = xcalloc(1, sizeof(*conn));
+    conn->ctx = NULL;
+    conn->conn_info = info;
+    conn->encryption_type = 'c';
+    /* Only public files (chmod o+r) accessible to non-root */
+    conn->uid = CF_UNKNOWN_OWNER;                    /* Careful, 0 is root! */
+    /* conn->maproot is false: only public files (chmod o+r) are accessible */
+
+    Log(LOG_LEVEL_DEBUG,
+        "New connection (socket %d).",
+        ConnectionInfoSocket(info));
+    return conn;
+}
+
+static void CFTestD_DeleteConn(ServerConnectionState *conn)
+{
+    int sd = ConnectionInfoSocket(conn->conn_info);
+    if (sd != SOCKET_INVALID)
+    {
+        cf_closesocket(sd);
+    }
+    ConnectionInfoDestroy(&conn->conn_info);
+
+    *conn = (ServerConnectionState) {0};
+    free(conn->session_key);
+    free(conn);
+}
+
+static void *CFTestD_HandleConnection(void *c)
+{
+    ServerConnectionState *conn = c;
+
+    char aligned_ipaddr[CF_MAX_IP_LEN + 2];
+
+    strlcpy(aligned_ipaddr, conn->ipaddr, sizeof(aligned_ipaddr));
+    strlcat(aligned_ipaddr, "> ",         sizeof(aligned_ipaddr));
+    /* Pad with enough spaces for IPv4 addresses to be aligned. Max chars are
+     * 15 for the address plus two for "> " == 17. */
+    size_t len;
+    for (len = strlen(aligned_ipaddr); len < 17; len++)
+    {
+        aligned_ipaddr[len] = ' ';
+    }
+    aligned_ipaddr[len] = '\0';
```

Comment:
This doesn't seem to be used anywhere.

Reply:
It is used in the log context.

Reply:
Or, that is what it is used for in cf-serverd. Removing it.